bubble_sort_by.rb

```diff
@@ -0,0 +1,28 @@
+def bubble_sort_by(array)
+	(array.length).times do |x|
+		(array.length - (x + 1)).times do |y|
+			if array[y + 1] != nil
+				case yield array[y], array[y + 1]
+				when 1
+					pivot = array[y]
+					array[y] = array[y + 1]
+					array[y + 1] = pivot
+				end
+			end
+		end
+	end
+end
+
+my_array = ['why', 'howdy', 'constitution', 'hi', 'whatup']
+
+puts "Unsorted array:"
+print my_array
+puts ''
+
+bubble_sort_by(my_array) do |x, y|
+	x.length <=> y.length
+end
+
+puts "Sorted array:"
+print my_array
+puts ''
```

Comment:
**Optional**. Instead of testing like this, you can start playing with RSpec testing. It makes your life easier.